### PR TITLE
added params for virtual ip in sdc resource

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add virtual IP support in SDC resource.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: powerflex/provider/sdc_resource_schema.go [hunk 1]
```diff
@@ -170,6 +170,30 @@ var sdcDetailSchema schema.ListNestedAttribute = schema.ListNestedAttribute{
...
+			},
+			"virtual_ip_nics": schema.StringAttribute{
+				MarkdownDescription: "The NIC to which the virtual IP addresses are mapped.",
+				Description:         "The NIC to which the virtual IP addresses are mapped.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"data_network_ip": schema.StringAttribute{
```
Comment (Krunal-Thakkar) [inline powerflex/provider/sdc_resource_schema.go:189]: Can we add validation when virtual IP is provided, this needs to provide??

Comment (shenda1) [reply powerflex/provider/sdc_resource_schema.go:191]: Added validation

**Comment by shenda1** [comment]
run e2e test

**Comment by rounak-adhikary** [comment]
Have we missed any test files somewhere? I see that more test variables have been added in provider_test.go, but I don't see any changes in actual tests.

**Comment by shenda1** [comment]
> Have we missed any test files somewhere? I see that more test variables have been added in provider_test.go, but I don't see any changes in actual tests.



> Have we missed any test files somewhere? I see that more test variables have been added in provider_test.go, but I don't see any changes in actual tests.



> Have we missed any test files somewhere? I see that more test variables have been added in provider_test.go, but I don't see any changes in actual tests.

The added variables are required when the cluster is configured with a virtual IP. The cluster where acceptance tests run is not configured with virtual ip.

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**